lib/tlIO/FFmpegWrite.cpp

```diff
@@ -109,14 +109,19 @@ namespace tl
             {
                 throw std::runtime_error(string::Format("{0}: Cannot allocate stream").arg(p.fileName));
             }
+            AVPixelFormat pix_fmt = AV_PIX_FMT_YUV420P;
```

Comment:
If there are no supported formats that seems like it should be an error. Setting it to AV_PIX_FMT_YUV420P works?

Reply:
No, not really.  It might just as welll set to AV_PIX_FMT_NONE.  I just thought of giving it one last try (and let ffmpeg fail by itself).
The only real problem is the pointer that comes as nullptr when you try to save H264 without libx264 which crashes Linux.

Reply:
In that case I think it might be better to throw an error like:
```
const AVCodec* avCodec = avcodec_find_encoder(avCodecID);
if (!avCodec)
{
    throw std::runtime_error(string::Format("{0}: Cannot find encoder").arg(p.fileName));
}
if (!avCodec->pix_fmts)
{
    throw std::runtime_error(string::Format("{0}: No pixel formats available").arg(p.fileName));
}
```

Or just put the nullptr check around the original assignment:
```
if (avCodec->pix_fmts)
{
    p.avCodecContext->pix_fmt = avCodec->pix_fmts[0];
}
```